packages/espressocash_app/lib/features/kyc_sharing/screens/email_confirmation_screen.dart

```diff
@@ -0,0 +1,97 @@
+import 'package:flutter/material.dart';
+
+import '../../../di.dart';
+import '../../../l10n/l10n.dart';
+import '../../../ui/bottom_button.dart';
+import '../../../ui/loader.dart';
+import '../../../ui/snackbar.dart';
+import '../services/kyc_service.dart';
+import '../utils/kyc_utils.dart';
+import '../widgets/kyc_page.dart';
+import '../widgets/kyc_text_field.dart';
+
+class EmailConfirmationScreen extends StatefulWidget {
+  const EmailConfirmationScreen({super.key});
+
+  static Future<bool> push(BuildContext context) => Navigator.of(context)
+      .push<bool>(
+        MaterialPageRoute(
+          builder: (context) => const EmailConfirmationScreen(),
+        ),
+      )
+      .then((result) => result ?? false);
+
+  @override
+  State<EmailConfirmationScreen> createState() =>
+      _EmailConfirmationScreenState();
+}
+
+class _EmailConfirmationScreenState extends State<EmailConfirmationScreen> {
+  final _controller = TextEditingController();
+
+  bool get _isValid => _controller.text.length == 6;
+
+  Future<void> _handleConfirm() async {
+    final success = await runWithLoader<bool>(
+      context,
+      () async {
+        try {
+          final service = sl<KycSharingService>();
+          await service.verifyEmail(code: _controller.text);
+
+          return true;
+        } on Exception {
+          if (!mounted) return false;
+          showCpErrorSnackbar(
```

Comment:
It's not necessarily wrong code, what if it's network exception?

Reply:
Updated error handling